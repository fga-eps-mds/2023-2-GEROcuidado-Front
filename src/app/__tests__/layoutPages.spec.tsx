import React from 'react';
import { render } from '@testing-library/react-native';
import PublicLayout from '../public/_layout';
import PrivateLayout from '../private/pages/_layout';

// Mock expo-router
jest.mock('expo-router', () => ({
  Stack: ({ children }: { children: React.ReactNode }) => <>{children}</>,
}));

describe('Layouts', () => {
  it('renders PublicLayout correctly', () => {
    const { toJSON } = render(<PublicLayout />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders PrivateLayout correctly', () => {
    const { toJSON } = render(<PrivateLayout />);
    expect(toJSON()).toMatchSnapshot();
  });
});