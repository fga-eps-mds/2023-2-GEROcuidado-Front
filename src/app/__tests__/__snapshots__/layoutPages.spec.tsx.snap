// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layouts renders PrivateLayout correctly 1`] = `
<View
  style={
    {
      "zIndex": 9999,
    }
  }
>
  <View
    collapsable={false}
    onLayout={[Function]}
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    pointerEvents="box-none"
    style={
      {
        "alignItems": "center",
        "justifyContent": "center",
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "transform": [
          {
            "translateY": 0,
          },
        ],
      }
    }
    testID="toastAnimatedContainer"
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#FFF",
          "borderLeftColor": "#69C779",
          "borderLeftWidth": 5,
          "borderRadius": 6,
          "elevation": 2,
          "flexDirection": "row",
          "height": 60,
          "opacity": 1,
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 6,
          "width": 340,
        }
      }
      testID="toastTouchableContainer"
    >
      <View
        style={
          [
            {
              "alignItems": "flex-start",
              "flex": 1,
              "justifyContent": "center",
              "paddingHorizontal": 25,
            },
            undefined,
          ]
        }
        testID="toastContentContainer"
      />
    </View>
  </View>
</View>
`;

exports[`Layouts renders PublicLayout correctly 1`] = `
<View
  style={
    {
      "zIndex": 9999,
    }
  }
>
  <View
    collapsable={false}
    onLayout={[Function]}
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    pointerEvents="box-none"
    style={
      {
        "alignItems": "center",
        "justifyContent": "center",
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "transform": [
          {
            "translateY": 0,
          },
        ],
      }
    }
    testID="toastAnimatedContainer"
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#FFF",
          "borderLeftColor": "#69C779",
          "borderLeftWidth": 5,
          "borderRadius": 6,
          "elevation": 2,
          "flexDirection": "row",
          "height": 60,
          "opacity": 1,
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 6,
          "width": 340,
        }
      }
      testID="toastTouchableContainer"
    >
      <View
        style={
          [
            {
              "alignItems": "flex-start",
              "flex": 1,
              "justifyContent": "center",
              "paddingHorizontal": 25,
            },
            undefined,
          ]
        }
        testID="toastContentContainer"
      />
    </View>
  </View>
</View>
`;
